docs/tutorial/plot_function_transformer.py

```diff
@@ -0,0 +1,193 @@
+"""
+Issues with FunctionTransformer
+===============================
+
+A pipeline including a `FunctionTransformer
+<https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.FunctionTransformer.html>`_
+cannot be automatically converted into onnx because there is no converter able to
+convert custom python code into ONNX. A custom converter needs to be written
+specifically for it.
+
+Initial try
++++++++++++
+
+A very simple pipeline and the first attempt to convert it into ONNX.
+"""
+import numpy as np
+from numpy.testing import assert_allclose
+from pandas import DataFrame
+from sklearn.base import TransformerMixin, BaseEstimator
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.preprocessing import FunctionTransformer
+from sklearn.compose import ColumnTransformer
+from sklearn.pipeline import Pipeline
+from skl2onnx import to_onnx
+
+# For the custom converter
+from skl2onnx import update_registered_converter
+from skl2onnx.common.utils import check_input_and_output_numbers
+from skl2onnx.algebra.onnx_ops import OnnxSlice, OnnxSub, OnnxDiv, OnnxMul, OnnxCastLike
+
+# To check discrepancies
+from onnx.reference import ReferenceEvaluator
+from onnxruntime import InferenceSession
+
+
+def calculate(df):
+    df["c"] = 100 * (df["a"] - df["b"]) / df["b"]
+    return df
+
+
+mapper = ColumnTransformer(
+    transformers=[
+        ("c", FunctionTransformer(calculate), ["a", "b"]),
+    ],
+    remainder="passthrough",
+    verbose_feature_names_out=False,
+)
+mapper.set_output(transform="pandas")
+
+pipe = Pipeline([("mapper", mapper), ("classifier", DecisionTreeClassifier())])
+
+data = DataFrame(
+    [
+        dict(a=1, b=2, f=5),
+        dict(a=4, b=5, f=10),
+    ]
+)
+y = np.array([0, 1], dtype=np.int64)
+pipe.fit(data, y)
+
+try:
+    to_onnx(pipe, data[:1], options={"zipmap": False})
+except Exception as e:
+    print("It does not work:", e)
+
+##################################
+# Use of custom transformer
+# +++++++++++++++++++++++++
+#
+# It is easier to write a custom converter if the FunctionTransformer
+# is implemented as a custom transformer.
+
+
+class OverpriceCalculator(BaseEstimator, TransformerMixin):
+    def __init__(self):
+        pass
+
+    def calculate_overprice(self, x, y):
+        return 100 * (x - y) / y
+
+    def fit(self, X, y=None):
+        return self
+
+    def transform(self, X, y=None):
+        x = X.apply(lambda x: self.calculate_overprice(x.a, x.b), axis=1)
```

Comment:
I would like to return all other columns (a,b,f) in the data frame along with the new column "c"

Reply:
I updated the pipeline to retain those two: `("ab", FunctionTransformer(), ["a", "b"]),`